plugins/httpapi/aci.py

```diff
@@ -0,0 +1,325 @@
+            msg = "Error on attempt to logout from APIC. {0}".format(exc_logout)
+            raise ConnectionError(self._return_info(None, method, path, msg))
+        self.connection._auth = None
+        self._verify_response(response, method, path, response_data)
+
+    def set_parameters(self):
+        connection_parameters = {}
+        for key in CONNECTION_KEYS:
+            value = self.params.get(key) if self.params.get(key) is not None else self.connection.get_option(CONNECTION_MAP.get(key, key))
+            self.connection.set_option(CONNECTION_MAP.get(key, key), value)
+            if key == "timeout" and self.params.get(key) is not None:
+                self.connection.set_option("persistent_connect_timeout", value + 30)
+
+            connection_parameters[key] = value
+            if self.connection_parameters and value != self.connection_parameters.get(key) and key in RESET_KEYS:
+                self.connection._connected = False
+                self.connection.queue_message("step", "Re-setting connection due to change in {0}".format(key))
+
+        if self.params.get("private_key") is not None:
+            self.connection.set_option("session_key", None)
+            connection_parameters["certificate_name"] = self.params.get("certificate_name")
+            connection_parameters["private_key"] = self.params.get("private_key")
+        elif self.connection.get_option("session_key") is not None and self.params.get("password") is None:
+            connection_parameters["certificate_name"] = list(self.connection.get_option("session_key").keys())[0]
+            connection_parameters["private_key"] = list(self.connection.get_option("session_key").values())[0]
+        else:
+            if self.connection_parameters.get("private_key") is not None:
+                self.connection._connected = False
+            self.connection.set_option("session_key", None)
+            connection_parameters["private_key"] = None
+            connection_parameters["certificate_name"] = None
+
+        if self.connection_parameters != connection_parameters:
+            self.connection_parameters = copy(connection_parameters)
+
+        self.set_hosts()
+
+    def set_hosts(self):
+         if self.params.get("host") is not None:
+            get_hosts = ast.literal_eval(self.params.get("host")) if "[" in self.params.get("host") else self.params.get("host").split(",") 
+         else:
+            if self.inventory_hosts is None:
+                self.inventory_hosts = re.sub(r"[[\]]", "", self.connection.get_option("host")).split(",")
+            get_hosts = self.inventory_hosts
+
+         if self.provided_hosts is None:
+            self.provided_hosts = deepcopy(get_hosts)
+            self.connection.queue_message(
+                "step:", "Provided Hosts: {0}".format(self.provided_hosts)
+            )
+            self.backup_hosts = deepcopy(get_hosts)
+            self.current_host = self.backup_hosts.pop(0)
```

Comment:
Add log message for the current host set.

Reply:
done